frame/support/src/storage/mod.rs

```diff
@@ -44,71 +47,121 @@ pub mod types;
 pub mod unhashed;
 pub mod weak_bounded_vec;
 
-#[cfg(all(feature = "std", any(test, debug_assertions)))]
-mod debug_helper {
-	use std::cell::RefCell;
+mod transaction_level_tracker {
+	use core::sync::atomic::{AtomicU32, Ordering};
+
+	type Layer = u32;
+	static NUM_LEVELS: AtomicU32 = AtomicU32::new(0);
```

Comment:
The native runtime can be executed from multiple threads.... 

Why don't we just have used the environmental crate then?

Reply:
It is hard to do here because there is no easy common function where to place it.

Reply:
@kianenigma had a solution that looked reasonable to me.